dev/devicelab/lib/framework/adb.dart

```diff
@@ -169,6 +200,21 @@ class AndroidDeviceDiscovery implements DeviceDiscovery {
     print('Device chosen: $_workingDevice');
   }
 
+  @override
+  Future<void> chooseWorkingDeviceById(String deviceId) async {
+    deviceId = _findMatchId(await discoverDevices(), deviceId);
+    if (deviceId != null) {
+      _workingDevice = AndroidDevice(deviceId: deviceId);
+      print('Choose device by ID: ' + _workingDevice.toString());
+    }
+    else {
```

Comment:
nit: I would early return to avoid the else

```
deviceId = _findMatchId(await discoverDevices(), deviceId);
if (deviceId != null) {
   _workingDevice = AndroidDevice(deviceId: deviceId);
    print('Choose device by ID: ' + _workingDevice.toString());
    return;
}
throw DeviceException(
  'Device with ID $deviceId is not found for operating system: '
  '$deviceOperatingSystem'
 );
```